python/lsst/galaxy_shear/processShearTest.py

```diff
@@ -0,0 +1,144 @@
+        self.centroidPlugin = None
+        self.measPlugin = None
+        for plugin in self.measurement.plugins.keys():
+            if plugin.find("Centroid") > 0:
+                self.centroidPlugin = self.measurement.plugins[plugin]
+            else:
+                self.measPlugin = self.measurement.plugins[plugin]
+
+    #   Transfer information already know about the sources from the GalSim epoch catalog
+    #   to an afwTable catalog.  The Psf library and index in particular are needed.
+    def buildSourceCatalog(self, imageBBox, dataRef):
+        """Build an empty source catalog, using the provided sim catalog's position to generate
+        square Footprints and its ID to set that of the source catalog.
+        """
+        sourceCat = lsst.afw.table.SourceCatalog(self.schema)
+        sourceCat.getTable().setMetadata(self.algMetadata)
+        simCat = dataRef.get("epoch_catalog", immediate=True)
+        xKey = simCat.schema.find('x').key
+        yKey = simCat.schema.find('y').key
+        idKey = simCat.schema.find('ID').key
+        nGals = imageBBox.getWidth() / self.config.galaxyStampSize
+        assert nGals * self.config.galaxyStampSize == imageBBox.getWidth()
+        n = imageBBox.getWidth() / nGals
+        self.dims = lsst.afw.geom.Extent2I(n, n)
+        offset = lsst.afw.geom.Extent2I(simCat[0][xKey], simCat[0][yKey])
+        for simRecord in simCat:
+            sourceRecord = sourceCat.addNew()
+            sourceRecord.setId(simRecord.get(idKey))
+            sourceRecord.set("psf_index", simRecord.get("psf_index"))
+            sourceRecord.set("psf_library", simRecord.get("psf_library"))
+            position = lsst.afw.geom.Point2I(simRecord.get(xKey), simRecord.get(yKey))
+            bbox = lsst.afw.geom.Box2I(position - offset, self.dims)
+            footprint = lsst.afw.detection.Footprint(bbox, imageBBox)
+            peakRecord = footprint.getPeaks().addNew()
+            peakRecord.setFx(position.getX())
+            peakRecord.setFy(position.getY())
+            sourceRecord.setFootprint(footprint)
+        return sourceCat
+
+    #   Run a measurement algorithm on the set of sources provided by
+    #   a GalSim catalog.  Also run a centroid algorithm if one is provided
+    def run(self, dataRef):
+        exposure = self.buildExposure(dataRef)
+        sourceCat = self.buildSourceCatalog(exposure.getBBox(lsst.afw.image.PARENT), dataRef)
+        images_file = dataRef.get("image", immediate=True)
+        #   the GalSim catalog will provide the number of the psf_library
+        #   and the index of the hdu with the psf for each source.
+        for source in sourceCat:
+            #   Locate the psb_library and get the indexed psf
+            dataRef.dataId["psf_index"] = source.get("psf_index")
+            dataRef.dataId["psf_library"] = source.get("psf_library")
+            psf_file = dataRef.get("psf_library", immediate=True)
```

Comment:
Yes, of course.  The whole reason for this design was to sort the catalog.  However, I put it off because an issue came up with the library naming in the work with Simon, and I decided not to do this until next Sprint.  It is not something I will forget.